### ISSUE TITLE
What is the mean of prediction's result?

### ISSUE THREAD

**Issue by wyqnumber** [dmlc/xgboost#306]
bst.predict(test)

I have three class labels. 0, 1, 2

I want to get labels like 0,1,2 rather than below result.

Thank you!

the result is:
[ 2.08266497 -0.16195059 -0.53449929]
 [ 2.08266497 -1.48556077  0.13058886]
 [ 1.32865083 -2.025424   -0.13586384]
 [-0.55171859 -1.54363155  1.26611865]
 [ 1.32865083 -2.025424   -0.13586384]
 [ 2.08266497 -0.70571303 -0.8340435 ]
 [ 1.3542099  -2.025424   -0.53449929]
 [ 2.08266497 -0.99530995 -0.8340435 ]
 [ 2.08266497 -0.99530995 -0.8340435 ]
 [ 3.17738867 -1.54363155  0.19758049]
 [ 2.08266497 -2.025424   -0.53449929]
 [ 1.57963073 -1.00376821 -0.53449929]
 [ 1.3542099  -1.27166677 -0.53449929]
 [ 2.62930393 -2.025424   -0.0407176 ]
 [ 2.62930393 -2.025424   -0.8340435 ]
 [ 2.08266497 -2.025424    0.13058886]
 [ 1.52446699 -2.18625474  0.75243491]
 [ 2.01449347 -1.27166677  0.07861483]
....

**Comment by tqchen** [comment]
If you want to get the label use multi:softmax as your objective, see https://github.com/dmlc/xgboost/blob/master/demo/multiclass_classification/train.py

**Comment by wyqnumber** [comment]
thanks! But, I run the program each time, the result is always changed. why? In order to get the best result?

**Comment by tqchen** [comment]
There could be indeterministic bahvior due to multithreading and float point rounding, see #9
